scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -651,6 +651,13 @@ def run_cmake(self, args="", filter_stages=None):
             f'-DPython3_EXECUTABLE={pathlib.Path(sys.executable).as_posix()}'
         ]
 
+        if self.instance.testsuite.harness == 'bsim':
+            cmake_args.extend([
```

Comment:
@ardo-nordic
How come did you add this here? (just because they were added in compile.sh ? or did somebody ask for them?

Reply:
Same arguments are added in zephyr/tests/bsim/compile.source by default.

Reply:
For context: I'm the one who added run_parallel and compile.sh, and I'm the one who added setting COVERAGE=y by default. I did so because at the time we were generating coverage reports from PRs.
But we haven't done so for a while, and beyond telling people how to generate them in the documentation, I don't think many people is using it for anything nowadays.
https://docs.zephyrproject.org/latest/develop/test/coverage.html
https://docs.zephyrproject.org/latest/develop/test/bsim.html#test-coverage-and-babblesim
So I have been feeling that building (and running) with coverage enabled has been a waste of CI time, and have been inclined to remove it for quite a while now. twister has it's own "--coverage" flag we instruct people to use if they want to generate a report.

Wrt ASSERTS I think we should have that set in the tests' conf files, or maybe in the board defconfig instead (@jori-nordic had set it in the compile.source in 1b4844aad0e0e49422b7f0a0ce92001113431319) (@Thalley @jhedberg @alwa-nordic feel free to chip in). I mean, I think it is good to set it for testing, I just don't think it is good to set it in hidden places.

The CMAKE_EXPORT_COMPILE_COMMANDS came bundled in
https://github.com/zephyrproject-rtos/zephyr/commit/f023b5f611df8f8fa975bf2ebd3e80239b4e9278
from the llcp refactoring topic branch. I don't think we ever really needed that set(?) (@kruithofa do you recall why this was done?)

Reply:
We should have asserts enabled by default for all bsim testing IMO. If there is a better way to set it by default (and not per test), then I'm OK with moving it.

Reply:
> We should have asserts enabled by default for all bsim testing IMO.

@Thalley With all bsim testing, are you thinking: a) "whenever we build targeting this board" or b) "whenever we run any sample or test in CI", c) "whenever we build any test (not sample) for this board", d) something else 
?